server/src/assets/game/script-game.js

```diff
@@ -796,6 +805,8 @@ function initialise() {
 		null                                      // trainId
 	);
 
+	const platform = getPlatformName();
```

Comment:
This will not suffice. The server will only return the platform name once `startup` has been called on it: the config file with the platform/module name that is returned is only parsed on startup. And this client-side initialization may run before startup has occured.

We will have to come up with some async try-and-retry logic, where this client will try to get the platform name again and again until it succeeds. Once it succeeds, the possible destinations and signal flag map needs to be updated. Luckily, users won't be able to grab any trains as long as startup has not occured, so we do not need any complicated locking logic.

Reply:
I've added a first attempt at solving this by using fetch and retrying on a non 200-code response. Needs testing.